core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java

```diff
@@ -94,19 +95,57 @@ public final class GeyserServer {
 
     private final GeyserImpl geyser;
     private EventLoopGroup group;
+    // Split childGroup may improve IO
+    private EventLoopGroup childGroup;
     private final ServerBootstrap bootstrap;
     private EventLoopGroup playerGroup;
 
     @Getter
     private final ExpiringMap<InetSocketAddress, InetSocketAddress> proxiedAddresses;
+    private final int listenCount;
 
-    private ChannelFuture bootstrapFuture;
+    private ChannelFuture[] bootstrapFutures;
 
     public GeyserServer(GeyserImpl geyser, int threadCount) {
         this.geyser = geyser;
-        this.group = TRANSPORT.eventLoopGroupFactory().apply(threadCount);
+        // bindCount = 1/2 of processors, and max to 20
+        // It's offers up to 20*100=2000 workload
+        int threadBindCount = this.geyser.getConfig().getThreadBindCount();
+        if (Bootstraps.isReusePortAvailable()) {
+            if (threadBindCount == -1) {
+                // Set to the minimum of 20 and threadCount if configThreadCount is -1
+                threadBindCount = Math.min(20, threadCount);
+                this.geyser.getLogger().info(String.format("Thread bind count is auto set to %s!", threadBindCount));
+            } else if (threadBindCount < 0) {
+                // Check for negative values and set to 1 if negative
+                this.geyser.getLogger().warning("Thread bind count in config cannot be negative, force set to 1!");
+                threadBindCount = 1;
+            } else {
+                // Ensure threadBindCount is the lesser of 20 or threadCount
+                if (threadBindCount > 20) {
+                    this.geyser.getLogger().warning("Thread bind count in config is bigger than 20, force set to 20!");
+                    threadBindCount = 20;
+                }
+                if (threadBindCount > threadCount) {
+                    threadBindCount = threadCount;
+                    this.geyser.getLogger().warning(String.format("Thread bind count in config is bigger than Netty's default thread, force set to %s!", threadBindCount));
+                }
+            }
+
+            this.listenCount = threadBindCount;
+        } else {
+            if (threadBindCount != 1) {
+                this.geyser.getLogger().warning("System does not support modify thread_bind_count; force to 1.");
+            }
+            this.listenCount = 1;
+        }
+//        this.listenCount = Bootstraps.isReusePortAvailable() ? Math.max(1, Math.min(20, threadCount / 4)) : 1;
+        this.group = TRANSPORT.eventLoopGroupFactory().apply(listenCount);
+        this.childGroup = TRANSPORT.eventLoopGroupFactory().apply(threadCount);
```

Comment:
Are there benefits to have different thread counts for the "parent" group and the listener? If yes, which count has to be used in the GeyserServer#bind() method?
Similarly, the config value you added allows people to just set the "listenCount", but not the threadCount (the latter can already be set here: https://github.com/GeyserMC/Geyser/blob/master/core/src/main/java/org/geysermc/geyser/GeyserImpl.java#L354-L361 using a system property).

Wouldn't it make sense to make the config option set that threadCount instead, and base the listen count on that? Or is there a benefit to having both separate? If there's a specific reason to split them, that's also fine; just want to avoid having two similar-ish config options (one of them even hidden away :/) that are both somewhat related.

Reply:
I've noticed that the configuration for threadCount was removed previously, and it's set to default based on the system's thread configuration. On a server, this value is generally quite large. As for listenCount, if it were configurable, it could be set to 1 to avoid wasting resources, as previously mentioned. However, if both parameters used the same configuration, it wouldn't be possible to optimize for both.

That being said, I believe that using the default settings for both listenCount and threadCount is sufficient. The maximum value of listenCount is only 20, and even for servers with a small number of players online, the performance impact is negligible.

In my opinion, these two parameters generally do not require additional configuration. For my personal server, I use the Geyser Bungee Plugin method, and both parameters are set to their default values. Up to now, I haven't encountered any problems.

Reply:
I agree with you that it shouldn't require additional configuration, yeah.

It should be configurable, but a config option is probably overkill - could you instead create a system property for the listen count?

See here for we've done it for the thread count:
https://github.com/GeyserMC/Geyser/blob/master/core%2Fsrc%2Fmain%2Fjava%2Forg%2Fgeysermc%2Fgeyser%2FGeyserImpl.java#L354-L361

Using a system property instead of a config option makes more sense here, as it is configurable for cases where it needs to be - e.g. high player counts with dedicated hardware - but shouldn't impact most users. I also agree with Tim that we shouldn't have a too high default value for the listen count,  since most servers wouldn't need that anyways.

tl;dr: Please remove the config option, and add a system property instead - and let's limit the default value as it's usually enough as is :)

Reply:
I'd add a new system property specifically for the listen threads so people can adjust both if need be

Reply:
yeah, that should work fine
Maybe worth printing the number of listen threads as a debug message via the logger, since that might be useful to see
Aka
`GeyserImpl.getInstance().getLogger().debug("Listen thread count:" + listen count);`